itest/lnd_funding_test.go

```diff
@@ -771,24 +820,28 @@ func testBatchChanFunding(ht *lntest.HarnessTest) {
 		},
 		OutputIndex: batchResp.PendingChannels[2].OutputIndex,
 	}
+	chanPoint4 := &lnrpc.ChannelPoint{
+		FundingTxid: &lnrpc.ChannelPoint_FundingTxidBytes{
+			FundingTxidBytes: batchResp.PendingChannels[3].Txid,
+		},
+		OutputIndex: batchResp.PendingChannels[3].OutputIndex,
+	}
 
+	// Mine the batch transaction and check the network topology.
 	block := ht.MineBlocksAndAssertNumTxes(6, 1)[0]
 	ht.Miner.AssertTxInBlock(block, txHash)
 	ht.AssertTopologyChannelOpen(alice, chanPoint1)
 	ht.AssertTopologyChannelOpen(alice, chanPoint2)
 	ht.AssertTopologyChannelOpen(alice, chanPoint3)
+	ht.AssertTopologyChannelOpen(alice, chanPoint4)
 
 	// Check if the change type from the batch_open_channel funding is P2TR.
 	rawTx := ht.Miner.GetRawTransaction(txHash)
-	require.Len(ht, rawTx.MsgTx().TxOut, 4)
-
-	// For calculating the change output index we use the formula for the
-	// sum of consecutive of integers (n(n+1)/2). All the channel point
-	// indexes are known, so we just calculate the difference to get the
-	// change output index.
-	changeIndex := uint32(6) - (chanPoint1.OutputIndex +
-		chanPoint2.OutputIndex + chanPoint3.OutputIndex)
+	require.Len(ht, rawTx.MsgTx().TxOut, 5)
 
+	// Since we have 4 channel outputs we expect the change output to be
+	// located at the fifth output index.
+	changeIndex := uint32(4)
```

Comment:
I don't think this is always the case, since we're using BIP-0069 to sort the inputs/outputs (if I'm not mistaken). So I think we need to keep the (weird but surprisingly working) formula above.

Reply:
The formula applied seems to always yield 4 as change index. I've restored the calculation and added an example.